test/successful_step.jl

```diff
@@ -0,0 +1,43 @@
+using Test
+using DynamicalSystemsBase
+using OrdinaryDiffEq: Vern9
+
+@testset "successful_step tests -> unstable systems" begin
+
+	############## define unstable dynamical rules ##################
+	#exp_rule_discrete(x, p, n) = SVector(p[1]*Base.exp(x[1]), p[2]*Base.exp(-x[2]))
+	exp_rule(x, p, t) = SVector(p[1]*Base.exp(x[1]), p[2]*Base.exp(-x[2]))
+	u0 = ones(2)
+	p0 = [1.0, 2.0]
+	
+	#discrete
+	dim = DeterministicIteratedMap(exp_rule, u0, p0) 
+	
+	#continuous
+	diffeq = (alg = Vern9(), abstol = 1e-9, reltol = 1e-9)
+	ode = CoupledODEs(exp_rule, u0, p0; diffeq)
+	
+	#projected
+	projection = [1]
+	complete_state = [0.0]
+	pr = ProjectedDynamicalSystem(ode, projection, complete_state)
```

Comment:
In this and all following creations of systems use `deepcopy(ode)`. Becuase otherwise you have a reference to the same system.

Reply:
Right. I had that but deleted them for some reason. I add them back in

Reply:
Btw can you explain the reason behind this? Why is the underlying `ds` passed by reference and not by value?

Reply:
in DynamicalSystems 3.0 all instances of dynamical systems are integrators: mutable entities that are stepped in place. Julia always passes mutable entities by reference.